x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/privileges/read_privileges_route.ts

```diff
@@ -4,60 +4,43 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import Hapi from 'hapi';
 import { merge } from 'lodash/fp';
 
+import { IRouter } from '../../../../../../../../../src/core/server';
 import { DETECTION_ENGINE_PRIVILEGES_URL } from '../../../../../common/constants';
-import { LegacyServices } from '../../../../types';
-import { RulesRequest } from '../../rules/types';
-import { GetScopedClients } from '../../../../services';
-import { transformError, getIndex } from '../utils';
+import { buildSiemResponse, transformError } from '../utils';
 import { readPrivileges } from '../../privileges/read_privileges';
 
-export const createReadPrivilegesRulesRoute = (
-  config: LegacyServices['config'],
-  usingEphemeralEncryptionKey: boolean,
-  getClients: GetScopedClients
-): Hapi.ServerRoute => {
-  return {
-    method: 'GET',
-    path: DETECTION_ENGINE_PRIVILEGES_URL,
-    options: {
-      tags: ['access:siem'],
-      validate: {
-        options: {
-          abortEarly: false,
-        },
+export const readPrivilegesRoute = (router: IRouter, usingEphemeralEncryptionKey: boolean) => {
+  router.get(
+    {
+      path: DETECTION_ENGINE_PRIVILEGES_URL,
+      validate: false,
+      options: {
+        tags: ['access:siem'],
       },
     },
-    async handler(request: RulesRequest, headers) {
+    async (context, request, response) => {
+      const siemResponse = buildSiemResponse(response);
       try {
-        const { clusterClient, spacesClient } = await getClients(request);
+        const clusterClient = context.core.elasticsearch.dataClient;
+        const siemClient = context.siem.getSiemClient();
 
-        const index = getIndex(spacesClient.getSpaceId, config);
-        const permissions = await readPrivileges(clusterClient.callAsCurrentUser, index);
-        return merge(permissions, {
-          is_authenticated: request?.auth?.isAuthenticated ?? false,
+        const index = siemClient.signalsIndex;
+        const clusterPrivileges = await readPrivileges(clusterClient.callAsCurrentUser, index);
+        const privileges = merge(clusterPrivileges, {
+          is_authenticated: true, // until we support optional auth: https://github.com/elastic/kibana/pull/55327#issuecomment-577159911
```

Comment:
@FrankHassanabad @XavierM I forgot to call this one out in my notes, sorry about that!

As mentioned in the comment, we currently don't have access to the user's auth state here. It's coming, and in the meantime I could thread the service through here, but this was the best "temporary" solution given that, in the current implementation, anonymous users shouldn't be able to reach this page.

My concern here is that this behavior breaks the existing auth-free demos that we use, but I need more info about how those work in order to maintain parity here.

Reply:
Look at the front end code it does expect to be able to get a valid value of being authenticated or not:

```
x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/helpers.tsx
```

```ts
export const redirectToDetections = (
  isSignalIndexExists: boolean | null,
  isAuthenticated: boolean | null,
  hasEncryptionKey: boolean | null
) =>
  isSignalIndexExists != null &&
  isAuthenticated != null &&
  hasEncryptionKey != null &&
  (!isSignalIndexExists || !isAuthenticated || !hasEncryptionKey);
```

So, yeah, we need a way to get that data so we can keep users out of the detection page if they are not authenticated. Saying they are authenticated when they are not is probably going to cause us issues.

Reply:
Looking at the demos we have such as: 
https://demo.elastic.co/

It looks like it has authenticated users fwiw:
<img width="367" alt="Screen Shot 2020-03-02 at 10 03 30 PM" src="https://user-images.githubusercontent.com/1151048/75744510-ae468980-5cd1-11ea-9e59-cab3c95c784d.png">

Reply:
If those demos don't function unless you're signed in, then I don't think we have an issue here, right? That boolean has never been `false` for the duration of the project, since there's no way to hit those routes without being signed in.

Reply:
Per our talks you can have unauthenticated users and we will have to create a ticket for us to fix this after this PR is merged from new platform.